internal/syncd/integrations/ecr/reconciler.go

```diff
@@ -2,27 +2,42 @@ package ecr
 
 import (
 	"context"
-
 	"ship-it/internal"
 
-	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
 )
 
-type RepositoriesService interface {
-	GetContents(ctx context.Context, org, repo, path string, opts *github.RepositoryContentGetOptions) (*github.RepositoryContent, []*github.RepositoryContent, *github.Response, error)
-	UpdateFile(ctx context.Context, owner, repo, path string, opt *github.RepositoryContentFileOptions) (*github.RepositoryContentResponse, *github.Response, error)
+type HelmReleaseEditor interface {
+	UpdateAndReplace(ctx context.Context, releaseName string, image *internal.Image) error
+}
+
+type IndexerService interface {
+	Lookup(repo string) ([]types.NamespacedName, error)
 }
 
 type ImageReconciler struct {
-	// TODO
+	RepoService  HelmReleaseEditor
+	IndexService IndexerService
 }
 
-func NewReconciler(org string, r RepositoriesService) *ImageReconciler {
+func NewReconciler(r HelmReleaseEditor, i IndexerService) *ImageReconciler {
 	return &ImageReconciler{
-		// TODO
+		RepoService:  r,
+		IndexService: i,
 	}
 }
 
 func (r *ImageReconciler) Reconcile(ctx context.Context, image *internal.Image) error {
+	releases, err := r.IndexService.Lookup(image.Repository)
+	if err != nil {
+		return errors.Wrapf(err, "failed to obtain the releases corresponding to the repository: %s", image.Repository)
+	}
+	for _, release := range releases {
+		err := r.RepoService.UpdateAndReplace(ctx, release.Name, image)
+		if err != nil {
```

Comment:
We should think more about the behaviour here. We would probably like to continue trying to update the rest of the releases and aggregate errors as we go (we could use the `syncd.multiError` type).

Let's use this logic for now and come back to it.

Reply:
For sure! Happy to open another PR adjusting this, I agree it is not a good behaviour for the entire update to fail as soon as an error is thrown.